features/vaadin-dashlets/dashlet-bsm/src/main/java/org/opennms/features/vaadin/dashboard/dashlets/BSMConfigurationWindow.java

```diff
@@ -0,0 +1,432 @@
+        ok.addClickListener(new Button.ClickListener() {
+            @Override
+            public void buttonClick(Button.ClickEvent event) {
+                if (!m_limitTextField.isValid()) {
+                    return;
+                }
+
+                m_dashletSpec.getParameters().put("filterByName", (m_filterByNameCheckBox.getValue() ? "true" : "false"));
+
+                if (m_filterByNameCheckBox.getValue()) {
+                    m_dashletSpec.getParameters().put("nameValue", m_nameTextField.getValue());
+                } else {
+                    m_dashletSpec.getParameters().put("nameValue", "");
+                }
+
+                m_dashletSpec.getParameters().put("filterByAttribute", (m_filterByAttributeCheckBox.getValue() ? "true" : "false"));
+
+                if (m_filterByAttributeCheckBox.getValue()) {
+                    m_dashletSpec.getParameters().put("attributeKey", m_attributeKeyTextField.getValue());
+                    m_dashletSpec.getParameters().put("attributeValue", m_attributeValueTextField.getValue());
+                } else {
+                    m_dashletSpec.getParameters().put("attributeKey", "");
+                    m_dashletSpec.getParameters().put("attributeValue", "");
+                }
+
+                m_dashletSpec.getParameters().put("filterBySeverity", (m_filterBySeverityCheckBox.getValue() ? "true" : "false"));
+
+                if (m_filterBySeverityCheckBox.getValue() && m_severitySelect.getValue() != null) {
+                    m_dashletSpec.getParameters().put("severityValue", m_severitySelect.getValue().toString());
+                } else {
+                    m_dashletSpec.getParameters().put("severityValue", OnmsSeverity.WARNING.getLabel());
+                }
+
+                if (m_filterBySeverityCheckBox.getValue() && m_compareOperatorSelect.getValue() != null) {
+                    m_dashletSpec.getParameters().put("severityCompareOperator", m_compareOperatorSelect.getValue().toString());
+                } else {
+                    m_dashletSpec.getParameters().put("severityCompareOperator", BusinessServiceSearchCriteriaBuilder.CompareOperator.GreaterOrEqual.name());
+                }
+
+                if (m_orderBy.getValue() != null) {
+                    m_dashletSpec.getParameters().put("orderBy", m_orderBy.getValue().toString());
+                } else {
+                    m_dashletSpec.getParameters().put("orderBy", BusinessServiceSearchCriteriaBuilder.Order.Name.name());
+                }
+
+                if (m_ascDesc.getValue() != null) {
+                    m_dashletSpec.getParameters().put("orderAsc", "Ascending".equals(m_ascDesc.getValue())?"true":"false");
+                } else {
+                    m_dashletSpec.getParameters().put("orderAsc", "true");
+                }
+
+                m_dashletSpec.getParameters().put("resultsLimit", m_limitTextField.getValue().toString());
```

Comment:
The generation of the spec looks wrong. 
Most of the code looks the same and should be generalized a bit more.